GirafAPI/Authorization/OrgAdminRequirement.cs

```diff
@@ -7,18 +11,25 @@ public class OrgAdminRequirement : IAuthorizationRequirement;
 public class OrgAdminAuthorizationHandler : AuthorizationHandler<OrgAdminRequirement>
 {
     private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly UserManager<GirafUser> _userManager;
 
-    public OrgAdminAuthorizationHandler(IHttpContextAccessor httpContextAccessor)
+    public OrgAdminAuthorizationHandler(IHttpContextAccessor httpContextAccessor, UserManager<GirafUser> userManager)
     {
         _httpContextAccessor = httpContextAccessor;
+        _userManager = userManager;
     }
 
     protected override Task HandleRequirementAsync(
         AuthorizationHandlerContext context,
         OrgAdminRequirement requirement)
     {
-        var claims = context.User;
-        var orgIds = claims.Claims
+        var userId = _userManager.GetUserId(_httpContextAccessor.HttpContext.User);
+        var user = _userManager.FindByIdAsync(userId).GetAwaiter().GetResult();
```

Comment:
Isn't it possible that this could be null?

Reply:
It's also aynsc so shouldn't the method be an async method to not block the thread?

Reply:
Good catch on the first question, I'll change the way the tokens contain claims so that it's only the ones actually used (userId and userName). It shouldn't be possible for it to be null since the tokens contain some default claims about identity.

I'll look into if the method should be async now

Reply:
The claims themselves live in a table on the server, earlier it was also then sending those claims in a token. That caused the issue with it not updating since the token was only generated on login. Now it just uses the token to get the userId to check the claims as they exist in the table so they're up to date